### DISCUSSION TITLE
Where to register a RegionBehavior when using Forms?

### DISCUSSION THREAD

**Discussion by jameslavery** [PrismLibrary/Prism#2786] [Xamarin.Forms]
I need to add a Region Behavior to dispose of Views and ViewModels when the Region closes them.

I've worked out how to code the Behavior itself, but all references to Region Behaviors mention overriding the Bootstrapper's `ConfigureDefaultRegionBehaviors` method to add the Behavior.

However, I can't find a Bootstrapper class to override. Is this available in Prism.Forms?

I'm using:

- Prism.DryIoc.Forms 8.0.0.1909
- Prism.Forms.Regions 8.0.0.1909

Or is there another way to achieve this without a RegionBehavior?

**Comment by brianlagunas** [comment]
https://prismlibrary.com/docs/xamarin-forms/behaviors/page-behavior-factory.html

**Reply by jameslavery** [reply]
@brianlagunas thanks - but that refers to **Page** Behaviors. I need to add a **Region** Behavior. Can I use the same pattern to apply my Region Behavior with a RegionBehaviorFactory?

**Reply by brianlagunas** [reply]
Oh sorry, I'm replying from my phone and missed that part. You'd probably have to resolve the IRegionBehaviorFactory and add your region behaviors to it.

However, I'm pretty sure the regions implement IDestructible already. Why not just use that?

**Reply by jameslavery** [reply]
Great - thank you. I was going to go down the path of resolving IRegionBehaviorFactory, so shows I was on the right path. Equally, as you say, it looks like you've already got IDestructibleRegionBehavior, so  it should 'just work'. I need to investigate more!
